### PR TITLE
Add migration guide

### LINKED ISSUE #1004: 1.0.0 migration guide
There are a handful of breaking changes. We should briefly document how to move from pre-1.0.0 to 1.0.0.

### PR DESCRIPTION
**Pull request by samanpwbb**
@katydecorah for review 

closes https://github.com/mapbox/assembly/issues/1004

### CODE REVIEW

File: MIGRATION_GUIDES.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+- `col--offr2` → `mr-1/6`
+- `col--offr3` → `mr-1/4`
+- `col--offr4` → `mr-1/3`
+- `col--offr5` → `mr-5/12`
+- `col--offr6` → `mr-1/2`
+- `col--offr7` → `mr-7/12`
+- `col--offr8` → `mr-2/3`
+- `col--offr9` → `mr-3/4`
+- `col--offr10` → `mr-5/6`
+- `col--offr11` → `mr-11/12`
```
Comment (katydecorah) [inline MIGRATION_GUIDES.md:74]: Is this list missing `col--offr12`?

File: MIGRATION_GUIDES.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+- `col--offl2` → `ml-1/6`
+- `col--offl3` → `ml-1/4`
+- `col--offl4` → `ml-1/3`
+- `col--offl5` → `ml-5/12`
+- `col--offl6` → `ml-1/2`
+- `col--offl7` → `ml-7/12`
+- `col--offl8` → `ml-2/3`
+- `col--offl9` → `ml-3/4`
+- `col--offl10` → `ml-5/6`
+- `col--offl11` → `ml-11/12`
```
Comment (katydecorah) [inline MIGRATION_GUIDES.md:85]: Is this list missing `col--offl12`?

File: MIGRATION_GUIDES.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+- `col--2` → `w-1/6`
+- `col--3` → `w-1/4`
+- `col--4` → `w-1/3`
+- `col--5` → `w-5/12`
+- `col--6` → `w-1/2`
+- `col--7` → `w-7/12`
+- `col--8` → `w-2/3`
+- `col--9` → `w-3/4`
+- `col--10` → `w-5/6`
+- `col--11` → `w-11/12`
```
Comment (katydecorah) [inline MIGRATION_GUIDES.md:62]: Is this list missing `col--12`?

**Comment by katydecorah** [comment]
I've been using this guide on dr-ui. It's working well! One question:

What do you think about a section to call out sizing and color changes? This would remind me to check these elements to make sure they work well with my page. 

Maybe something like:

## Changes to element sizing, color, and defaults

- The height of `toggle` is now 30px, down from 36px and the height of `toggle-s` is now 18px, down from 24px.
- The default form control colors is gray, originally blue, for all form types.
- The typographic scale has been updated. Notably `txt-m` is now 16px, up from 15px and many headings have a smaller font-size.
- Shadow classes now have a y-offset.
- The default `select` is now borderless. To add a border, use `select--stroke`.

**Comment by samanpwbb** [comment]
@katydecorah i pulled in those changes. Back to you again!

**Review by katydecorah — APPROVED** [review]
Thanks for your patience in my review! This guides works great. I left a few non-blocking questions.

**Pull request merged**